src/rust/generic-subgraph-generator/src/serialization.rs

```diff
@@ -0,0 +1,21 @@
+use serde::Deserialize;
+use sqs_lambda::event_decoder::PayloadDecoder;
+use std::io::Cursor;
+
+#[derive(Debug, Clone, Default)]
+pub struct ZstdJsonDecoder;
+
+impl<D> PayloadDecoder<D> for ZstdJsonDecoder
+where
+    for<'a> D: Deserialize<'a>,
+{
+    fn decode(&mut self, body: Vec<u8>) -> Result<D, Box<dyn std::error::Error>> {
+        let mut decompressed = Vec::new();
```

Comment:
Minor, but I would assume that anything decompressed will be at least the size of its input, so we should just do `Vec::with_capacity(body.len())` here I think

Reply:
Good call. I'll adjust.